src/fal/cli/dbt_runner.py

```diff
@@ -166,3 +165,76 @@ def _create_fal_result_file(target_path: str, run_index: int):
         shutil.copy(
             fal_run_result, os.path.join(target_path, f"fal_results_{run_index}.json")
         )
+
+
+# This is the Python implementation of the `dbt_run()` function, in which
+# we directly use dbt-core as a Python library. We don't run it directly
+# but rather use 'multiprocessing' to run it in a real system Process to
+# imitate the existing behavior of `dbt_run()` (in terms of performance).
+
+# dbt-core is currently using the spawn as its mulitprocessing context
+# so we'll mirror it.
+multiprocessing.set_start_method("spawn", force=True)
+
+
+def _dbt_run_through_python(args: List[str], target_path: str, run_index: int):
+    from dbt.main import handle_and_check
+
+    try:
+        run_results, success = handle_and_check(args)
+    except BaseException as exc:
+        run_results = None
+        return_code = getattr(exc, "code", 1)
+        traceback.print_exc()
+    else:
+        return_code = 0 if success else 1
+
+    logger.debug(f"dbt exited with return code {return_code}")
+
+    # The 'run_results' object has a 'write()' method which is basically json.dump().
+    # We'll dump it directly to the fal results file (instead of first dumping it to
+    # run results and then copying it over).
+    if run_results is not None:
+        run_results.write(os.path.join(target_path, f"fal_results_{run_index}.json"))
+    else:
+        raise RuntimeError("Error running dbt run")
+
+    return return_code
+
+
+def dbt_run_through_python(
+    args: argparse.Namespace, models_list: List[str], target_path: str, run_index: int
+) -> DbtCliOutput:
+    """Run DBT from the Python entry point in a subprocess."""
+    args = get_dbt_command_list(args, models_list, include_cli_only_args=False)
+    args_as_text = " ".join(args)
+    logger.info(f"Running DBT with these options: {args_as_text}")
+
+    # We are going to use multiprocessing module to spawn a new
+    # process that will run the sub-function we have here. We
+    # don't really need a process pool, since we are only going
+    # to spawn a single process but it provides a nice abstraction
+    # over retrieving values from the process.
```

Comment:
What it does is make a "container" for running the function? That is what "abstraction" means here?

Reply:
> That is what "abstraction" means here?

Normally it is hard to [retrieve the return value](https://docs.python.org/3/library/multiprocessing.html#sharing-state-between-processes) from a `multiprocessing.Process`. The `multiprocessing.Pool` however offers a nice abstraction over this problem and let you directly use the return value from a function.